x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/reindex/flyout/progress.tsx

```diff
@@ -124,95 +132,168 @@ const ReindexingDocumentsStepTitle: React.FunctionComponent<{
 };
 
 const orderedSteps = Object.values(ReindexStep).sort() as number[];
-const getStepTitle = (step: ReindexStep, inProgress?: boolean): ReactNode => {
+
+const getStepTitle = (
+  step: ReindexStep,
+  meta: ReindexState['meta'],
+  inProgress?: boolean
+): ReactNode => {
   if (step === ReindexStep.readonly) {
     return inProgress ? (
       <FormattedMessage
         id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.reindexingChecklist.inProgress.readonlyStepTitle"
-        defaultMessage="Setting original index to read-only."
+        defaultMessage="Setting {indexName} index to read-only."
+        values={{
+          indexName: <EuiCode>{meta.indexName}</EuiCode>,
+        }}
       />
     ) : (
       <FormattedMessage
         id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.reindexingChecklist.readonlyStepTitle"
-        defaultMessage="Set original index to read-only."
+        defaultMessage="Set {indexName} index to read-only."
+        values={{
+          indexName: <EuiCode>{meta.indexName}</EuiCode>,
+        }}
       />
     );
   }
+
   if (step === ReindexStep.newIndexCreated) {
     return inProgress ? (
       <FormattedMessage
         id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.reindexingChecklist.inProgress.createIndexStepTitle"
-        defaultMessage="Creating new index."
+        defaultMessage="Creating {reindexName} index."
+        values={{
+          reindexName: <EuiCode>{meta.reindexName}</EuiCode>,
+        }}
       />
     ) : (
       <FormattedMessage
         id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.reindexingChecklist.createIndexStepTitle"
-        defaultMessage="Create new index."
+        defaultMessage="Create {reindexName} index."
+        values={{
+          reindexName: <EuiCode>{meta.reindexName}</EuiCode>,
+        }}
       />
     );
   }
+
   if (step === ReindexStep.aliasCreated) {
     return inProgress ? (
       <FormattedMessage
-        id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.reindexingChecklist.inProgress.aliasSwapStepTitle"
-        defaultMessage="Swapping original index with alias."
+        id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.checklistStep.reindexingChecklist.inProgress.aliasCreatedStepTitle"
+        defaultMessage="Creating {indexName} alias for {reindexName} index."
```

Comment:
WDYT about using i18n pluralization feature in order to avoid the ternary and all the duplication that it conveys?